src/solver/conv_hip_implicit_gemm_bwd_data_xdlops.cpp

```diff
@@ -0,0 +1,391 @@
+                                                              nullptr,
+                                                              args.N,
+                                                              args.K,
+                                                              args.C,
+                                                              args.input,
+                                                              args.filter,
+                                                              args.output,
+                                                              args.strides,
+                                                              args.dilation,
+                                                              args.lPadding,
+                                                              args.rPadding,
+                                                              {},
+                                                              {},
+                                                              {});
+        if(conv_ptrs[i]->IsSupportedArgument(argument_ptr.get()))
+            return true;
+    }
+    return false;
+}
+
+template <typename DataType>
+void ConvHipImplicitGemmBwdXdlops::RunCKSolution(
+    const Handle& handle,
+    const AnyInvokeParams& primitive_parameters,
+    const ProblemDescription& problem,
+    const PerformanceConfigHipImplicitGemmBwdXdlops& config) const
+{
+    const auto args      = CKArgsBwd{problem};
+    const auto conv_ptrs = DeviceOpBwdPtrs<DataType>::GetInstances();
+    auto& conv_ptr       = conv_ptrs.at(config.index);
+    auto& data_ctx       = primitive_parameters.CastTo<conv::DataInvokeParams>();
+    const auto& tensors  = data_ctx.tensors;
+    auto argument_ptr    = conv_ptr->MakeArgumentPointer(
+        static_cast<void*>(tensors.out),
+        const_cast<void*>( // NOLINT (cppcoreguidelines-pro-type-const-cast)
+            static_cast<const void*>(tensors.w)),
+        const_cast<void*>( // NOLINT (cppcoreguidelines-pro-type-const-cast)
+            static_cast<const void*>(tensors.in)),
+        args.N,
+        args.K,
+        args.C,
+        args.input,
+        args.filter,
+        args.output,
+        args.strides,
+        args.dilation,
+        args.lPadding,
+        args.rPadding,
+        {},
+        {},
+        {});
+    auto invoker_ptr            = conv_ptr->MakeInvokerPointer();
+    const auto enable_profiling = handle.IsProfilingEnabled();
+
+    float elapsed_time =
+        invoker_ptr->Run(argument_ptr.get(), {handle.GetStream(), enable_profiling});
+    if(enable_profiling)
+    {
+        handle.ResetKernelTime();
+        handle.AccumKernelTime(elapsed_time);
+    }
+}
+#endif
+
+void PerformanceConfigHipImplicitGemmBwdXdlops::HeuristicInit(const ProblemDescription& problem)
+{
+    this->index = 0;
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = problem;
+#else
+    this->index      = 0;
+    this->total_size = 0;
```

Comment:
Is it ok to `this->total_size = 0;` even if `Init` fails?

Reply:
No, it should be -1. I will delete it. Thanks for pointing this out.

Reply:
@iq136boy Please also check Fwd counterpart of this solver. It may use the same pattern.

[Recommendation] Implement the fixes for `ConvHipImplicitGemmFwdXdlops` (if any) in a separate PR.

/cc @JehandadKhan

Reply:
> @iq136boy Please also check Fwd counterpart of this solver. It may use the same pattern.
> 
> [Recommendation] Implement the fixes for `ConvHipImplicitGemmFwdXdlops` (if any) in a separate PR.
> 
> /cc @JehandadKhan

@atamazov Sure. I will implement the fixes for ConvHipImplicitGemmFwdXdlops in a separate PR.